File: XUCore/localization/NSMenuLocalizationSupport.m
// 
// NSMenuLocalizationSupport.m
// 
// Created by Charlie Monroe
// 
// Copyright (c) 2010-13 Fuel Collective, LLC. All rights reserved.
// 


#import "NSMenuLocalizationSupport.h"

#ifndef FCLocalizedString
	#import "FCLocalizationSupport.h"
#endif

@implementation NSMenu (NSMenuLocalizationSupport)
-(void)localizeMenu{
	[self setTitle:FCLocalizedString([self title])];
	for (NSMenuItem *item in [self itemArray]){
		if ([item attributedTitle] != nil){
			NSAttributedString *attributedTitle = [item attributedTitle];
			NSString *localizedTitle = FCLocalizedString([attributedTitle string]);
			NSAttributedString *localizedAttributedTitle = [[NSAttributedString alloc] initWithString:localizedTitle attributes:[attributedTitle attributesAtIndex:0 effectiveRange:NULL]];
			[item setAttributedTitle:localizedAttributedTitle];
		}else{
			[item setTitle:FCLocalizedString([item title])];
		}
		if ([item hasSubmenu]){
			[[item submenu] localizeMenu];
		}
	}
}
@end

File: XUCore/localization/NSMenuLocalizationSupport.swift
//
//  NSMenuLocalizationSupport.swift
//  XUCore
//
//  Created by Charlie Monroe on 11/7/15.
//  Copyright © 2015 Charlie Monroe Software. All rights reserved.
//

import Foundation


public extension NSMenu {
	
	public func localizeMenu(_ bundle: Bundle = Bundle.main) {
		self.title = XULocalizedString(self.title)
		for item in self.items {
			if let attributedTitle = item.attributedTitle {
				let localizedTitle = XULocalizedString(attributedTitle.string, inBundle: bundle)
				let localizedAttributedTitle = NSAttributedString(string: localizedTitle, attributes: attributedTitle.attributes(at: 0, effectiveRange: nil))
				item.attributedTitle = localizedAttributedTitle
			}else{
				item.title = XULocalizedString(item.title, inBundle: bundle)
			}
			
			if item.hasSubmenu {
				item.submenu?.localizeMenu()
			}
		}
	}
	
}

